capsules/src/process_console.rs

```diff
@@ -170,6 +176,49 @@ impl Default for WriterState {
     }
 }
 
+#[derive(Copy, Clone)]
+pub struct Command {
+    buf: [u8; COMMAND_BUF_LEN],
+    len: usize,
+}
+impl Command {
+    pub fn new() -> Command {
+        Command {
+            buf: [0; COMMAND_BUF_LEN],
+            len: 0,
+        }
+    }
+
+    /// Fill the buffer with the provided data.
+    /// If the provided data's length is smaller than the buffer length,
+    /// the left over bytes are not modified due to '\0' termination.
+    pub fn fill(&mut self, buf: &[u8], terminator_idx: usize) {
+        self.len = if terminator_idx >= COMMAND_BUF_LEN {
+            COMMAND_BUF_LEN
+        } else {
+            terminator_idx
+        };
+
+        self.buf[..self.len].copy_from_slice(&buf[..self.len]);
+    }
+
+    pub fn is_buffer_empty(&mut self) -> bool {
+        self.len == 0
+    }
+
+    pub fn get_byte(&self, byte_idx: usize) -> Option<u8> {
+        if byte_idx >= self.len {
+            return None;
+        }
+
+        Some(self.buf[byte_idx])
+    }
+
+    pub fn same_bytes(&mut self, buf: &[u8; COMMAND_BUF_LEN]) -> bool {
```

Comment:
What about implementing the [`PartialEq`](https://doc.rust-lang.org/std/cmp/trait.PartialEq.html) trait? This allows you to use `==`.

Reply:
https://doc.rust-lang.org/std/cmp/trait.PartialEq.html#impl-PartialEq%3C%5BB%5D%3E-for-%5BA%5D
No need to manually implement it, simply adding it to derive would work.

Reply:
Can we implement the PartialEq for the Command structure, because in the implementation we compare the Command structure with an array not with another Command Structure, that means that in order to replace the `same_bytes` function we would need to add a dummy Command Structure:
```RUST
let mut dummy_command: Command = Command::new();
dummy_command.fill(&command_array, terminator);

if cmd_arr[0] != dummy_command {
// ....
}
```

Reply:
If we add simply to the derive wouldn't it compare a Command Structure with another Command structure, because we need to compare the Command with an array, in order not to add a dummy Command Structure